### ISSUE TITLE
Build the parser in terms of parser-combinators

### ISSUE THREAD

**Issue by Kleidukos** [Kleidukos/boreal#13]
The parser should use https://flora.pm/packages/@hackage/parser-combinators

**Comment by jameshaydon** [comment]
It looks like currently the parser is implemented purely in terms of `tree-sitter`, which is interesting. Is there some problem with this approach, hence this issue?

**Comment by Kleidukos** [comment]
@jameshaydon Nope, the parser is actually a combination of tree-sitter, for the production of a CST, an elaboration phase that has yet to exist, and then there is a home-made abomination: https://github.com/Kleidukos/boreal/blob/main/compiler/Boreal/IR/RawCore.hs

**Comment by jameshaydon** [comment]
Thanks for the clarification. So I suppose that means you are happy with `tree-sitter` for the `Text -> CST` parsing?

**Comment by Kleidukos** [comment]
oh yeah this rules
